### DISCUSSION TITLE
How to Achieve a Web Page Layout with Specific Percentage and Pixel Divisions?

### DISCUSSION THREAD

**Discussion by Daniel-Fei** [zauberzeug/nicegui#2734] [Q&A] [answered]
### Question

I have nearly completed the overall framework of the webpage. However, when constructing the main content, I wish to divide this section into three parts: the middle occupies 90% of the webpage width (as indicated by the red line in the diagram), and each side occupies 5% (as indicated by the blue lines in the diagram). This setup is so that I can add specific content to the middle part, such as text, images, tabs, etc. I've tried some methods but have not been successful in implementing the desired page layout. Could you advise on how to achieve this layout?

Additionally, if I want to change the proportions of the page to specific pixel values, such as setting the middle area to a width of 700px, the left side to 100px, and the right side to 200px, how should I modify it? Thank you!

![image](https://github.com/zauberzeug/nicegui/assets/34276117/2c75789d-c34a-4b57-bb07-05c35ee080a0)

**Comment by rodja** [comment]
Something like this should work:

```py
ui.query('.nicegui-content').classes('p-0')
with ui.row().classes('gap-0  w-full'):
    with ui.column().classes('w-[5%] bg-red'):
        ui.label('left')
    with ui.column().classes('w-[90%] bg-yellow'):
        ui.label('center')
    with ui.column().classes('w-[5%] bg-red'):
        ui.label('right')
```

> Additionally, if I want to change the proportions of the page to specific pixel values, such as setting the middle area to a width of 700px, the left side to 100px, and the right side to 200px, how should I modify it? Thank you!

I do not quite understand. Would you like to programatically change the size of the browser window? In your example the overall width must be 1000px.

**Comment by Daniel-Fei** [comment]
Thank you for your response. I tried the code you suggested, and it does work, but there seems to be an issue.
The width of the content does not occupy the total width of the page; it appears to only correspond to the width of this particular tab I guess?
What I actually want is for the width to span the total width of the page.  I'm not sure how to modify the code to make this row occupy the total width of the page.
I have uploaded a screenshot of the current state of the page, as well as the original code for this part.
Looking forward to your further guidance. Thank you very much!


![code](https://github.com/zauberzeug/nicegui/assets/34276117/4ad7f96f-5496-4a28-b1dd-9df300cb8323)
![homepage](https://github.com/zauberzeug/nicegui/assets/34276117/0195eb3e-4647-4cd2-9c4f-c2250476f7f2)

**Reply by falkoschindler** [reply]
@Daniel-Fei For some child element to occupy the whole page, its parent element should do so as well. You probably have to add something like `.classes('w-full')` to the tab panel.
In  general: Please don't post screenshots of source code, because it doesn't allow copy-pasting the code. Use Markdown code blocks instead.

**Comment by Daniel-Fei** [comment]
Thank you for the quick response. Here is the source code:

```py
router = Router()

ui.add_head_html('''
                <style>
                    :root {
                        --q-dark-page: #222222; /* Define a custom CSS variable for a dark page background color. */
                    }
                </style>
                ''')

with ui.header(elevated=False).style('background-color: #3e6a94; height: 100px').classes('items-center justify-between'):
    # Create a tabbed interface. Each tab will contain different parts of the patient's medical record.
    with ui.tabs() as tabs:
        # Create a 'Summary' tab with an emergency contact icon to display an overview of the patient's information.
        one = ui.tab('summary', label='Summary', icon='o_contact_emergency')
        
        # Create an 'Admission' tab with a post add icon for entering or viewing patient's admission notes.
        two = ui.tab('admission_note', label='Admission', icon='post_add')
        
        # Create a 'Progress' tab with a text snippet icon for tracking and displaying the patient's progress notes.
        three = ui.tab('progress_note', label='Progress', icon='o_text_snippet')
        
        # Create a 'Diagnosis' tab with a healing icon to show diagnostic information and conclusions.
        four = ui.tab('diagnosis_info', label='Diagnosis', icon='healing')
    
    add_navigation_dropdown_elements.main(router)

for ind, patient in enumerate(patients):
            
    @router.add(f"/{patient['name']}")
    def generate_patient_page(patient=patient): 
        with ui.tab_panels(tabs, value=one).classes('w-full'):
            with ui.tab_panel(one):
                ui.query('.nicegui-content').classes('p-0')
                
                with ui.row().classes('gap-0  w-full'):
                    with ui.column().classes('w-[15%] bg-red'):
                        ui.label('left')
                    with ui.column().classes('w-[70%] bg-yellow'):
                        ui.label('center')
                    with ui.column().classes('w-[15%] bg-red'):
                        ui.label('right')

            with ui.tab_panel(two):
                content_summary.main(patient)  
            with ui.tab_panel(three):
                content_summary.main(patient)  
            with ui.tab_panel(four):
                content_summary.main(patient)
```

**Reply by falkoschindler** [reply, marked as answer]
@Daniel-Fei Have you tried my suggestion?
> You probably have to add something like `.classes('w-full')` to the tab panel.

**Reply by Daniel-Fei** [reply]
Yes, I have tried, but it doesn't work. The width has not changed to occupy the whole page

**Reply by falkoschindler** [reply]
Ok. If it didn't solve your problem, I'm not sure why you marked this Q&A as answered. 🤔

In case you need further assistance, I'd like to ask you to provide a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) so that we can help more efficiently. Thanks!

**Comment by Daniel-Fei** [comment]
I've solved the problem by adding the following code to ui.tab_panels, thank you very much for all the help, thanks!

`.style("position: absolute; width: 100%; left: 50%; top: 5px; transform: translateX(-50%); border: none; background-color: transparent;")`
